### ISSUE TITLE
container.pack doesn't work well

### ISSUE THREAD

**Issue by jasongrout** [ipython/ipywidgets#126]
This example from the Widget Styling example notebook doesn't work:

```python
buttons = [widgets.Button(description=str(i)) for i in range(3)]
display(*buttons)
container = widgets.HBox(children=buttons)
display(container)
container.width = '100%'
container.pack = 'center'
```

Is supposed to center the buttons.  We should fix it or change the example.

**Comment by SylvainCorlay** [comment]
Might be solved with #124 .

**Comment by jdfreder** [comment]
> Might be solved with #124 .

\- maybe, but my understanding was that the widget area isn't filling the notebook anymore.  Is that right @jasongrout ?

**Comment by jasongrout** [comment]
It's the widget-subarea.  If I set the widget-subarea to 100% width, it works great

**Comment by jasongrout** [comment]
The above example doesn't work on master.

**Comment by jasongrout** [comment]
Problem is that setting widget-subarea to 100% width likely has drastic consequences.

**Comment by jdfreder** [comment]
I think we should remove the example, and fix this bug in our 4.1 release.

**Comment by jdfreder** [comment]
Hmm, something we've done since August has fixed this.  I just checked in master and the buttons are centered now with the code you provide.  Do you want to confirm?  Make sure to clear your cache too.
